### PR TITLE
Use erc1319 registry manifest

### PR DESCRIPTION
**Pull request by njgheorghita**
Update ethpm.js to use erc1319 manifest.

### CODE REVIEW

File: src/registries/test/registry.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	const manifestURI = await ethpm.registries.getReleaseData("ens", "1.0.0");
+	expect(manifestURI).toEqual("ipfs://QmeooZzPrT2hDWSkhGoyLeSecsSbU26E6RiYfkXfPoug9U");
+  });
+
+  it("can get a packages' releases", async() => {
+	const packageName = 'ethregistrar'
+	const releases = await Promise.all(
+	  Array.from(await ethpm.registries.package("wallet").releases())
+	);
+	expect(releases).toEqual(['1.0.0'])
```
Comment (njgheorghita) [inline src/registries/test/registry.ts]: ping @gnidan - I'm having hard time figuring out what I'm doing wrong here. `releases` always returns to be an empty array. This pr's a little pruned down, but locally I have it talking to the registry and grabbing the correct data, but I just can't figure out why I can't access that data via the `package().releases()` method. Any clues as to what I'm missing here?

File: src/registries/web3/cursors/packages.ts [hunk 2]
```diff
@@ -17,94 +17,45 @@ export default class PackagesCursor extends Paged<BN> implements IterableIterato
...
-      }));
-
-      return {
-        done: true,
-        value: promise,
-      };
+      return this.getName();
     }
+    return {
+      done: true,
```
Comment (gnidan) [inline src/registries/web3/cursors/packages.ts]: Why is this `true` now?

Not sure what happened to the pagination here, but I'd been keen to make sure we didn't break intended behavior! Let me know!

Comment (njgheorghita) [reply src/registries/web3/cursors/packages.ts]: Yeah - I did some refactoring here, but it's all tested so nothing should be broken. Originally, I couldn't get either of the `Cursor`s to return anything except for empty arrays (see the other comment in this pr). Essentially, i.e. `PackagesCursor` it was generated all the promises to fetch package names before it had retrieved the relevant package ids. So, I split those into two steps - 1. fetch all the package ids (which now happens in the web3 registry service) 2. fetch the corresponding package names via the cursor

> Why is this true now?

Some of the refactoring was based on this, which I saw everywhere
```js
{
  done: false | true // Iteration done or not
  value: T // value of iteration. May not be present when done is true
}
```

So basically, the tests are working, so everything seems to be working fine. But, if I missed something or if there is a better design options -  I'm all ears.

Comment (gnidan) [reply src/registries/web3/cursors/packages.ts]: Hm, I'll have to circle back here to evaluate the tests, but I guess my first question is: what happens if you have more packages than the page limit in the registry? Do the tests cover that?

I didn't write the original implementation so I'm not sure about what the behavior used to be.

Also note: TypeScript 3.6 introduced much better types for generators/iterators; it's likely worth upgrading to make use of those.

Comment (njgheorghita) [reply src/registries/web3/cursors/packages.ts]: Yup, that test case is covered [here](https://github.com/ethpm/ethpm.js/pull/30/files#diff-512d0016943411014c5e07692e3df9ebR50). But, I do plan to extend testing with a much larger registry soon.

good to know about the generators/iterators, I'll look into switching over to use them.
